torch/csrc/cuda/Module.cpp

```diff
@@ -225,6 +225,24 @@ PyObject * THCPModule_cudaCachingAllocator_raw_delete(PyObject *_unused, PyObjec
   END_HANDLE_TH_ERRORS
 }
 
+PyObject * THCPModule_getAllocatorBackend(PyObject *_unused, PyObject *noargs)
+{
+  HANDLE_TH_ERRORS
+  using c10::cuda::CUDACachingAllocator::AllocatorBackend;
+  AllocatorBackend backend = c10::cuda::CUDACachingAllocator::allocatorBackend();
+  // this call should be uncommon, don't bother interning strings
+  switch (backend) {
+    case AllocatorBackend::NATIVE:
+      return THPUtils_packString("native");
+    case AllocatorBackend::CUDAMALLOCASYNC:
+      return THPUtils_packString("cudaMallocAsync");
+    default:
+      THPUtils_assert(false, "Unexpected value for backend");
```

Comment:
It'd be helpful to print out the actual int value of backend since this should really be impossible